File: python/219. Contains Duplicate II.py
#219. Contains Duplicate II.py
"""
Given an array of integers and an integer k, find out whether there are two distinct indices i and j in the array such that nums[i] = nums[j] and the absolute difference between i and j is at most k.
"""
class Solution:
    def containsNearbyDuplicate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """

        dic = {}
        for i, v in enumerate(nums):
            if v in dic and i - dic[v] <= k:
                return True
            dic[v] = i
        return False    

File: C++/219. Contains Duplicate II.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map<int,int>store;
        for(int i = 0; i<nums.size();++i){
            if(store.find(nums[i])!=store.end() && i - store[nums[i]] <=k)
                return true;
            else
                store[nums[i]] = i;
        }
        return false;
    }
};